### DISCUSSION TITLE
C/C++ API for Gymnasium Mujoco RL tasks?

### DISCUSSION THREAD

**Discussion by skellco** [google-deepmind/mujoco#2103] [Asking for Help]
### Intro

Hi!
I'm a researcher developing RL algorithms in C/C++.

### My setup

We would like to evaluate C/C++ RL algorithms in the Gymnasium Mujoco continuous control benchmarks: https://gymnasium.farama.org/environments/mujoco/

### My question

Are there example C/C++ APIs for these environments?

### Minimal model and/or code that explain my question

If you encountered the issue in a complex model, please simplify it as much as possible (while still reproducing the issue).

Model:
<details>
<summary>minimal XML</summary>

```XML

<mujoco>
  <worldbody>
    <light pos="0 0 1"/>
    <geom type="sphere" size="1" rgba="1 0 0 1"/>
  </worldbody>
</mujoco>

```
</details>

Code:
```python
import mujoco
import mediapy as media
model = mujoco.MjModel.from_xml_string(xml)
data = mujoco.MjData(model)

with mujoco.Renderer(model) as renderer:
  mujoco.mj_forward(model, data)
  renderer.update_scene(data)

  media.show_image(renderer.render())
```

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by yuvaltassa** [comment]
Maybe take a look at envpool?

https://github.com/sail-sg/envpool
